lib/identity/client.js

```diff
@@ -67,6 +68,60 @@ class Client extends PartialClient {
     options.headers.Authorization = `Bearer ${token}`
     return fetch(url, options)
   }
+
+  async createSecret(secret) {
+    const trimmedName = secret.secretName.trim()
+    const trimmedValue = secret.secretValue.trim()
+    if (trimmedName === '') {
+      throw { message: 'Name cannot be empty' }
+    }
+    const re = new RegExp('^[a-zA-Z0-9-_]{1,30}$')
+    if (!re.test(trimmedName)) {
+      throw {
+        message: 'Secret name must contain alphanumeric characters, -, or _',
+      }
+    }
+    if (trimmedValue === '') {
+      throw { message: 'Value cannot be empty' }
+    }
+    const groupName = [`tozny.secret.${this.storage.config.clientId}`]
+    // check if the clients secret group already exists
+    let groups = await this.storage.listGroups(
+      this.storage.config.clientId,
+      groupName
+    )
+    let group
+    let groupMembers = []
+    if (groups.groups.length < 1) {
+      // create the group if it doesn't exist
+      let newGroup = await this.storage.createGroup(groupName[0])
+      group = newGroup.group
+      let capabilities = {
+        read: true,
+        share: true,
+      }
+      var member = new GroupMember(this.storage.config.clientId, capabilities)
```

Comment:
nit: either `let` or `const` is preferable for variable scoping. I generally use `const` first unless I need mutation, but I would just switch this to `let` to match the rest of the file.